pipeline/scripts/ci/upstream_cli.py

```diff
@@ -14,37 +13,63 @@
 import yaml
 from docopt import docopt
 from packaging import version
+from yaml.loader import SafeLoader
 
 LOG = logging.getLogger(__name__)
 
 
-def update_upstream(file_name, content):
+def set_lock(file_name):
+    """Method to lock the file to avoid race condition"""
+    lock_file = file_name + ".lock"
+    exit_status = os.system(f"ls -l {lock_file}")
+    if exit_status:
+        print("File lock does not exist, Creating it")
+        os.system(f"touch {lock_file}")
+        return
+    # If any process locks the file it will wait till it unlock
+    timeout = 600
+    timeout_start = time.time()
+    while time.time() < timeout_start + timeout:
+        # wait for 2 minutes before retrying
+        time.sleep(120)
+        exit_status = os.system(f"ls -l {lock_file}")
+        if exit_status:
+            print("File lock does not exist, Creating it")
+            os.system(f"touch {lock_file}")
+            return
+
+    raise Exception(f"Lock file: {lock_file} already exist, can not create lock file")
+
+
+def unset_lock(file_name):
+    """Method to unlock the above set lock"""
+    lock_file = file_name + ".lock"
+    exit_status = os.system(f"rm -f {lock_file}")
+    if exit_status == 0:
+        print("File un-locked successfully")
+        return
+    raise Exception("Unable to unlock the file")
+
+
+def update_upstream(file_name, content, ceph_version):
     """Method to update for the provided file
     along with file set lock to avoid race condition.
     Args:
         file_name: Name of the file to update
         content: things to update in file
     """
-    # waiting for 10 minutes in case lock is held indefinitely
-    timeout = 600
-    timeout_start = time.time()
-    while time.time() < timeout_start + timeout:
+    with open(file_name, "w") as yaml_file:
         try:
-            # try/except in case the file is still locked by another process
-            # open the file for editing
-            with open(file_name, "w") as yaml_file:
-                # To lock the file to avoid race condition
-                fcntl.flock(yaml_file, fcntl.LOCK_EX | fcntl.LOCK_NB)
-                # update the file with write operation
-                yaml_file.write(yaml.dump(content, default_flow_style=False))
-                # unlock it so other processed can use it
-                fcntl.flock(yaml_file, fcntl.LOCK_UN)
-                LOG.info(f"Updated build info in {file_name}")
-                break
-
-        except IOError:
-            # wait for 2 minutes before retrying
-            time.sleep(120)
+            # To lock the file to avoid race condition
+            set_lock(file_name)
+            # update the file with write operation
+            yaml_file.write(yaml.dump(content, default_flow_style=False))
+        finally:
+            # unlock it so other processed can use it
+            unset_lock(file_name)
+            LOG.info(
```

Comment:
Add build information in the log record.

Reply:
@sunilkumarn417 Thanks for the review It's done,Please review it.

> [sangadi@sangadi ci]$ sudo python3 upstream_cli.py build pacific
> 2022-06-08 13:26:38,053 - INFO - upstream_cli.py:143 - Upstream repo for the given branch is https://1.chacra.ceph.com/repos/ceph/pacific/3942eb53d5956599736edc406f34c95de1667e88/centos/8/flavors/default/repo
> 2022-06-08 13:26:38,053 - INFO - upstream_cli.py:148 - Upstream image for the given branch is quay.ceph.io/ceph-ci/ceph:3942eb53d5956599736edc406f34c95de1667e88
> podman version 3.4.2
> Trying to pull quay.ceph.io/ceph-ci/ceph:3942eb53d5956599736edc406f34c95de1667e88...
> Getting image source signatures
> Copying blob d5c7970ff9ce skipped: already exists  
> Copying blob e043f1f2adf0 skipped: already exists  
> Copying blob c08cdfc8cc65 skipped: already exists  
> Copying blob 36183769f78d skipped: already exists  
> Copying blob 8c0d5ebd25cf skipped: already exists  
> Copying config 1502e2568e done  
> Writing manifest to image destination
> Storing signatures
> 1502e2568e19f719ba2cebcab66b561e50c3ebfa496834ad2b5d0977347d5bd4
> 2022-06-08 13:26:45,601 - INFO - upstream_cli.py:172 - Upstream version for the given branch is 16.2.9-125
> ls: cannot access '/ceph/cephci-jenkins/latest-rhceph-container-info/upstream.yaml.lock': No such file or directory
> File lock does not exist, Creating it
> File un-locked successfully
> 2022-06-08 13:26:48,983 - INFO - upstream_cli.py:71 - Updated build info: 
>  repo:https://1.chacra.ceph.com/repos/ceph/pacific/3942eb53d5956599736edc406f34c95de1667e88/centos/8/flavors/default/repo 
>  image:quay.ceph.io/ceph-ci/ceph:3942eb53d5956599736edc406f34c95de1667e88 
>  ceph version:16.2.9-125 in /ceph/cephci-jenkins/latest-rhceph-container-info/upstream.yaml
>